src/combined-selectors/reporting-config-search-results.ts

```diff
@@ -54,16 +55,100 @@ function searchReportingConfig(
 		}
 	}
 
+	// Finally, search for the search term in the description of the products and feature areas
+	const searchTermTokens = tokenizeAndNormalize( searchTerm );
+	const scores: Record< string, number > = {};
+
+	for ( const token of searchTermTokens ) {
+		const matchingEntities = invertedIndex.get( token ) || [];
+
+		for ( const { id, weight } of matchingEntities ) {
+			if ( ! scores[ id ] ) {
+				scores[ id ] = 0;
+			}
+			scores[ id ] += weight;
+		}
+	}
+
+	const scoreThreshold = 1;
+
+	for ( const entityId in scores ) {
+		if ( scores[ entityId ] >= scoreThreshold ) {
+			if ( products[ entityId ] ) {
+				searchResults.products.add( entityId );
+			} else if ( featureGroups[ entityId ] ) {
+				searchResults.featureGroups.add( entityId );
+				searchResults.products.add( featureGroups[ entityId ].productId );
+			} else if ( features[ entityId ] ) {
+				searchResults.features.add( entityId );
+				if ( features[ entityId ].parentType === 'product' ) {
+					searchResults.products.add( features[ entityId ].parentId );
+				} else {
+					searchResults.featureGroups.add( features[ entityId ].parentId );
+					const parentFeatureGroup = featureGroups[ features[ entityId ].parentId ];
+					searchResults.products.add( parentFeatureGroup.productId );
+				}
```

Comment:
**Nit (non-blocking)**: The logic here for adding an entity and then correctly adding its parents is now repeated in this bigger search function -- how do you feel about making a little inline closure function to DRY it up a bit?

Something like...
```typescript
const addFeatureGroupAndParents( featureGroupId: string ) => {
    	searchResults.featureGroups.add( featureGroupId );
	searchResults.products.add( featureGroups[ featureGroupId ].productId );
}

const addFeatureAndParents( featureId: string ) => {
    searchResults.features.add( featureId );
    if ( features[ featureId ].parentType === 'product' ) {
	    searchResults.products.add( features[ featureId ].parentId );
    } else {
	    searchResults.featureGroups.add( features[ featureId ].parentId );
	    const parentFeatureGroup = featureGroups[ features[ featureId ].parentId ];
	    searchResults.products.add( parentFeatureGroup.productId );
    }
}

// ... then elsewhere ...
if ( products[ entityId ] ) {
    searchResults.products.add( entityId );
} else if ( featureGroups[ entityId ] ) {
    addFeatureGroupAndParents( entityId );
} else if ( features[ entityId ] ) {
    addFeatureAndParents( entityId );
}
```